cvmfs/compression.h

```diff
@@ -36,6 +37,55 @@ enum StreamStates {
   kStreamEnd,
 };
 
+// Do not change order of algorithms.  Used as flags in the catalog
+enum Algorithms {
+  kZlibDefault = 0,
+  kNoCompression,
+  kAny,
+};
+
+/** 
+  * Abstract Compression class which is inhereited by implementations of 
+  * compression engines such as zlib...
+  */
+class Compressor {
+  public:
+    Compressor() {};
+    virtual ~Compressor() {};
+    virtual int Deflate(upload::CharBuffer &outbuf, size_t &outbufsize, 
+            const unsigned char* inbuf, const size_t inbufsize, 
+            const bool flush) { return 0; };

```

Comment:
This can safely be pure virtual, or does a concrete `Compressor` without a `::Deflate()` make sense?